### ISSUE TITLE
New Printer

### ISSUE THREAD

**Issue by sambonuruddeen** [mike42/escpos-php#592]
I have used the code with this printer model Xprinter_XP-365B

**Comment by mike42** [comment]
Thanks for reporting. Were you able to print images on the printer via the `bitImage()` method?

**Comment by diegoletelierr** [comment]
> I have used the code with this printer model Xprinter_XP-365B

How did you manage to print? I installed the drivers and executed the initial code (Hello World), it manages to put the document in the queue of impressions but it does not print anything. Thanks.

![image](https://user-images.githubusercontent.com/34931453/48320675-1cf26600-e5fb-11e8-90d8-33a18e62808a.png)

**Comment by diegoletelierr** [comment]
> > I have used the code with this printer model Xprinter_XP-365B
> 
> How did you manage to print? I installed the drivers and executed the initial code (Hello World), it manages to put the document in the queue of impressions but it does not print anything. Thanks.
> 
> ![image](https://user-images.githubusercontent.com/34931453/48320675-1cf26600-e5fb-11e8-90d8-33a18e62808a.png)

Solved! 

![image](https://user-images.githubusercontent.com/34931453/48322799-14ecf300-e608-11e8-9eba-07c8f3c695a5.png)

**Comment by sambonuruddeen** [comment]
Try this code
<?php
require __DIR__ . '/vendor/mike42/escpos-php/autoload.php';



//use Mike42\Escpos\PrintConnectors\FilePrintConnector;

use Mike42\Escpos\Printer;
use Mike42\Escpos\PrintConnectors\WindowsPrintConnector;


try {

	include 'php_action/core.php';

$productId = $_GET['productId'];

//Test if barcode exists

$bcSQL = "SELECT product_name, barcode FROM product WHERE product_id=$productId";
$bcQR = $connect->query($bcSQL);
$bcRES = $bcQR->fetch_assoc();

	// Enter the share name for your USB printer here
	$connector = new WindowsPrintConnector("Xprinter_XP-365B");
	$printer = new Printer($connector);

	//* Barcodes */
//$barcodes = Printer::BARCODE_UPCA;

$printer -> setBarcodeHeight(80);

	$printer -> feed(1);
	$printer -> text(ucfirst($bcRES['product_name']).": ".$bcRES['barcode'] . "\n");
	$printer -> setJustification(Printer::JUSTIFY_CENTER);
	$printer -> barcode($bcRES['barcode']);
	$printer -> feed(10);

$printer -> cut();
$printer -> close();

} catch(Exception $e) {
	echo "Couldn't print to this printer: " . $e -> getMessage() . "\n";
}

**Comment by icehaunter** [comment]
Hello! Did you get it to print labels or only receipts?

**Comment by sambonuruddeen** [comment]
I only printed receipts.

**Comment by steph6u** [comment]
Hi, I am using Xprinter XP-365B on Windows 10. I am using the windows-usb.php sample code with my shared printer name replaced in the code. Script runs without notice or error, but printer does not print anything. Running copy script in command prompt also does not print. No errors. Please help. Thanks.

**Comment by sambonuruddeen** [comment]
Did you install the Xprinter driver?

**Comment by steph6u** [comment]
Yes, but I have also uninstalled it to use Generic / Text Only as per Mike's docs. Windows test print works on xprinter, but not on the Generic / Text Only driver. I see it come up split second on the printer queue, then gone, but nothing printed. Thanks for taking the time.

**Comment by xMoho** [comment]
@sambonuruddeen Hi, did you manage to cut receipts with that printer? The function cut() doesn't work in my case. (Without any error code) 
I also tried setting _cut after print_ from printer driver with no luck.

**Comment by sambonuruddeen** [comment]
> @sambonuruddeen Hi, did you manage to cut receipts with that printer? The function cut() doesn't work in my case. (Without any error code)
> I also tried setting _cut after print_ from printer driver with no luck.

I was not able to  cut receipts because the printer I used doesn't support receipt printing

**Comment by xMoho** [comment]
> > @sambonuruddeen Hi, did you manage to cut receipts with that printer? The function cut() doesn't work in my case. (Without any error code)
> > I also tried setting _cut after print_ from printer driver with no luck.
> 
> I was not able to cut receipts because the printer I used doesn't support receipt cutting

Ouch, you are totaly right... this printer does not have auto-cutting. 😞
It was my bad for not reading the entire product description so I deserve it. 😅 

Thank you so much!

**Comment by MatthK** [comment]
> 
> 
> Yes, but I have also uninstalled it to use Generic / Text Only as per Mike's docs. Windows test print works on xprinter, but not on the Generic / Text Only driver. I see it come up split second on the printer queue, then gone, but nothing printed. Thanks for taking the time.

I have the exact same problem. I have installed the printer driver, and tried the Hello World. It quickly shows up in the print queue, but nothing gets printed, and no error message shows up. I can print a test print from the printer preferences page though.
Did you ever get it to work?

**Comment by steph6u** [comment]
> > Yes, but I have also uninstalled it to use Generic / Text Only as per Mike's docs. Windows test print works on xprinter, but not on the Generic / Text Only driver. I see it come up split second on the printer queue, then gone, but nothing printed. Thanks for taking the time.
> 
> I have the exact same problem. I have installed the printer driver, and tried the Hello World. It quickly shows up in the print queue, but nothing gets printed, and no error message shows up. I can print a test print from the printer preferences page though.
> Did you ever get it to work?

Unfortunately, no. I haven't been able to get back to it. It would be great if someone can share a solution for this. My printer has just been sitting since.

**Comment by Koxal76** [comment]
i has same troubles (no errors and no print). But when i set switches 1=OFF, 2=ON, the printer work.
